service/api_task.go

```diff
@@ -386,7 +386,9 @@ func assignNextAvailableTask(taskQueue *model.TaskQueue, taskQueueService model.
 				"spec_group_max_hosts": spec.GroupMaxHosts,
 			}))
 
-			return nil, fmt.Errorf("cannot find a db.tasks document for the next task '%s' to be assigned to host '%s'", queueItem.Id, currentHost.Id)
+			// return nil, fmt.Errorf("cannot find a db.tasks document for the next task '%s' to be assigned to host '%s'", queueItem.Id, currentHost.Id)
+			// STU: Why do we not pass back the error in this scenario?
```

Comment:
Apparently there can be a task on the queue that isn't in the db https://jira.mongodb.org/browse/EVG-6214. I'm not sure that we ever figured out why though.